pygeoapi/api.py

```diff
@@ -868,6 +868,13 @@ def get_collection_item(self, headers_, format_, dataset, identifier):
         try:
             LOGGER.debug('Fetching id {}'.format(identifier))
             content = p.get(identifier)
+        except ProviderItemNotFoundError:
```

Comment:
Should this error go **after** the `ProviderConnectionError` below for logical sequence?